bundles/org.openhab.voice.actiontemplatehli/README.md

```diff
@@ -0,0 +1,384 @@
+Those are not required, as you can use the build-in white space or simple tokenizers (from OpenNLP),they are just required to use the match by lemmas and the optional language tag functionalities.
+
+There are some examples at the end that you can review if you want a general idea of what can be done.
+
+## NLP Terminology
+
+I will briefly explain some terms that will be used:
+
+* Tokenize: first step of the recognition is to split the text, each of these parts is called token. 
+* Named Entity Recognition (NER): is the process of finding a subset of tokens on the input.
+* Part Of Speech (POS) tagging: categorizing tokens in a text, depending on the definition of the token and its context.
+* Lemmatize: is the process of getting a generic representation of the tokens, each of it is called lemma. (Example of one token to lemma conversion: 'is' -> 'be').
+
+## Action Template Target:
+
+This interpreter allows two ways to target items:
+
+* You can link an action to a specific item by adding the custom metadata namespace 'actiontemplatehli' to it.
+* You can link an action to all items of a type by providing the file '<OPENHAB_USERDATA>/actiontemplatehli/type_actions/<ITEM_TYPE_NAME>.json' (here you can restrict each action by item tags).
+
+Two important notes:
+
+* Actions linked to items have prevalence over actions linked to a whole item type.
+* On actions linked to an item type the itemLabel placeholder will always be applied (explained bellow). If there are multiple item labels detected on the input the actions linked to both will be scored, each using the correct itemLabel value, so there should be no collisions in case a template contains a token that matches an item label.
+
+## Action Template Scoring
+
+The scoring mechanism scores each action configuration to look for the best scored.
+If a token fails the comparison, the action scores 0.
+If all actions scored 0, none is executed.
+
+The action configuration field 'type' defines if the template should be compared using tokens or lemmas.
+Please note that the captured placeholder value is extracted from the tokens not from the lemmas, but the equivalent lemmas are replaced before scoring.
+
+The action template string is a list of tokens separated by white space.
+You can use the ';' separator to provide alternative templates and the '|' to provide alternative tokens.
+Here is an example of string: "what app|application is open on $itemLabel;what app|application is on $itemLabel".
+
+Take in account that, as this is a token basis comparison, matching depends on the tokenizer you are using as they can produce different tokens for the same text.
+
+## Action Template Options:
+
+The location were action configurations are placed changes whetter you are targeting an item or many, so take a look to the 'Action Template Target' to understand were to put this configurations.
```

Comment:
The location **where** configurations ... **whether** you are ... to understand **where** to put ...